main.go

```diff
@@ -18,18 +39,69 @@ func getPing(w http.ResponseWriter, r *http.Request) {
 
 }
 
-func getBooks(w http.ResponseWriter, r *http.Request) {
+func Books(w http.ResponseWriter, r *http.Request) {
+	metodo := r.Method
+	fmt.Printf("metodo: %v\n", metodo)
+	switch {
+	case metodo == "GET":
+		getBooks(w, r)
+	case metodo == "POST":
+		postBooks(w, r)
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+	}
+}
 
-	bookslist := []struct { //Struct created to handle data of the list of books
-		Name      string  `json:"name"`
-		Price     float32 `json:"price"`
-		Inventory int     `json:"inventory"`
-	}{
-		{"Book 1", 30.20, 2},
-		{"Book 2", 20.30, 1},
-		{"Book 3", 32.20, 5},
+func postBooks(w http.ResponseWriter, r *http.Request) {
+	//TO DO:
+	//Read the Json body
+	var newBook = Book{"sem nome", -1, -1, uuid.Nil}
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&newBook)
+	if err != nil {
+		fmt.Println("error:", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	fmt.Printf("newBook: %+v\n", newBook)
+	fmt.Printf("bookslist: %+v\n", bookslist)
+
+	//Verify if the entry is in a valid format
+	switch {
+	case newBook.Name == "sem nome":
+		w.WriteHeader(http.StatusPartialContent)
+		w.Write([]byte("Insira um nome para cadastrar o livro."))
+	case newBook.Price == -1: //-1 means that no price was added
+		w.WriteHeader(http.StatusPartialContent)
+		w.Write([]byte("Insira um preço para cadastrar o livro."))
+	case newBook.Inventory == -1: //-1 menas that no inventory was added
+		w.WriteHeader(http.StatusPartialContent)
+		w.Write([]byte("Insira a quantidade deste livro em estoque."))
+	default:
+	}
```

Comment:
In this case you probably want to go with individual `if`s for each check.
This is not partial content, this is a `BadRequest`.
```go
if newBook.Name == "" {
        // do this
}

Reply:
Ok. I'll find a way of returning all possible warning messages at once.

Reply:
Done.

//Verify if the entry is in a valid format
	blankFields := ""
	if newBook.Name == "" {
		blankFields += "Insira um nome para cadastrar o livro.\n"
	}
	if newBook.Price == nil {
		blankFields += "Insira um preço para cadastrar o livro.\n"
	}
	if newBook.Inventory == nil {
		blankFields += "Insira a quantidade deste livro em estoque.\n"
	}
	if blankFields != "" {
		w.WriteHeader(http.StatusBadRequest)
		w.Write([]byte(blankFields))
		return
	}

Reply:
I,ve changed the order too. 
First: Check if already exists a book with the same name;
Second: Check if all fields are filled.